scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+                currentKey = entry.getKey();
+                Serializable value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                           BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Then)" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+                } else {
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(elseRemappingFunction,
+                                                                                                                                                           BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Else)" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public boolean waitUntil(String originator, TaskId taskid, String channel, String key, String predicate)
+            throws InvalidChannelException, CompilationException {
+        Channel chosenChannel = getChannel(channel);
+        Serializable value = chosenChannel.get(key);
+        try {
+            return evaluateClosure(predicate, BiPredicate.class).test(key, value);
+        } catch (CompilationException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public PredicateActionResult waitUntilThen(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)